test/functional/merge.py

```diff
@@ -0,0 +1,142 @@
+# merge.py
+#
+# A set of classes needed to create a vcdMerge specific test.
+#
+# Copyright (c) 2017 vcdMaker team
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+
+import os
+
+from flat import Flat, CommonFlat
+
+
+class Source(object):
+    """A class for storing a source specific parameters."""
+
+    def __init__(self, node, test_directory):
+        """The Source class constructor.
+
+        Arguments:
+        node - The XML node to be read.
+        test_directory - The test directory.
+        """
+        self.parameters = {'format': ['', 'Missing source format.'],
+                           'time_stamp': ['', 'Missing source time stamp.'],
+                           'time_unit': ['', 'Missing source time unit.'],
+                           'prefix': ['', ''],
+                           'line_counter': ['', ''],
+                           'input_file': ['', 'Missing source input file.']}
+
+        self.data = Flat(node, self.parameters)
+        self.source = []
+        self.create_source(test_directory)
+
+    def create_source(self, test_directory):
+        """Builds the list of source parameters."""
+        self.source = [self.data.get_parameter('format'), ',',
```

Comment:
Why are all of the ',' here? They could just be added by `join` in function `get` below.
```python
','.join(['a', 'b', 'c'])
```
Results in "a,b,c".